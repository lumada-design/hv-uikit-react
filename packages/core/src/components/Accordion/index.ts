export { default as accordionClasses } from "./accordionClasses";
export * from "./accordionClasses";
export * from "./Accordion";
