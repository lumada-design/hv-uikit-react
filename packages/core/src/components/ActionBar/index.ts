export { default as actionBarClasses } from "./actionBarClasses";
export * from "./actionBarClasses";
export * from "./ActionBar";
