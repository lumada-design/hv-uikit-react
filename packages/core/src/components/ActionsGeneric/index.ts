export { default as actionsGenericClasses } from "./actionsGenericClasses";
export * from "./actionsGenericClasses";
export * from "./ActionsGeneric";
