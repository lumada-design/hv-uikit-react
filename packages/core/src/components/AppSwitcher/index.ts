export * from "./AppSwitcher";
export { HvAction as HvAppSwitcherAction } from "./Action";
