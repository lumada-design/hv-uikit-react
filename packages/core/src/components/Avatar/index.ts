export { default as avatarClasses } from "./avatarClasses";
export * from "./avatarClasses";
export * from "./Avatar";
