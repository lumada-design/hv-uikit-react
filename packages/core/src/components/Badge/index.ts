export { default as badgeClasses } from "./badgeClasses";
export * from "./badgeClasses";
export * from "./Badge";
