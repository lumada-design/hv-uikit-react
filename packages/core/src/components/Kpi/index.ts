export { default as kpiClasses } from "./kpiClasses";
export * from "./kpiClasses";
export * from "./Kpi";
