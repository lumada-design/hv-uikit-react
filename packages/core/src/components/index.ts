export * from "./Box";
export { Box as HvBox } from "./Box";
export * from "./Button";
export { Button as HvButton } from "./Button";
export * from "./Dropdown";
export { Dropdown as HvDropdown } from "./Dropdown";
export * from "./Header";
export {
  Actions as HvHeaderActions,
  Brand as HvHeaderBrand,
  Header as HvHeader,
  Navigation as HvHeaderNavigation,
} from "./Header";
export * from "./Typography";
export { Typography as HvTypography } from "./Typography";
