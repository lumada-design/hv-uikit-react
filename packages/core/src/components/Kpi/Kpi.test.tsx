import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { TopXS } from "@hitachivantara/uikit-react-icons";

import { HvKpi } from "@core/components";

describe("Kpi", () => {
  it("should render all components", () => {
    render(
      <HvKpi
        indicatorTextVariant="title1"
        indicatorUnitTextVariant="title2"
        labels={{
          title: "Title",
          indicator: "Indicator",
          unit: "Unit",
          comparisonIndicatorInfo: "Comparison indicator info",
        }}
        visualComparison="Visual comparison"
        trendIndicator={<TopXS role="img" title="Trend indicator" />}
        visualIndicator={<TopXS role="img" title="Visual indicator" />}
      />
    );

    expect(screen.getByText("Title")).toBeInTheDocument();
    expect(screen.getByText("Indicator")).toBeInTheDocument();
    expect(screen.getByText("Unit")).toBeInTheDocument();
    expect(screen.getByText("Comparison indicator info")).toBeInTheDocument();
    expect(screen.getByText("Visual comparison")).toBeInTheDocument();
    expect(
      screen.getByRole("img", { name: "Trend indicator" })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("img", { name: "Visual indicator" })
    ).toBeInTheDocument();
  });
});
