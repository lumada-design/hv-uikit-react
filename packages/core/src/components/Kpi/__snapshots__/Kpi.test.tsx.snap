// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kpi > should render correctly 1`] = `
<div>
  <div
    class="HvKpi-root css-1p3o5cs-StyledRoot e70l5p69"
  >
    <div>
      <label
        class="HvTypography-root HvTypography-label css-93trrd-getStyledComponent e1tnpalo0"
      >
        Title
      </label>
    </div>
    <div
      class="HvKpi-indicatorsContainer css-1a6w73h-StyledIndicatorsContainer e70l5p68"
    >
      <h1
        class="HvKpi-spacingToTheRight HvKpi-indicatorText e70l5p66 HvTypography-root HvTypography-title1 css-1agro9v-getStyledComponent-StyledIndicatorText e1tnpalo0"
      >
        Indicator
      </h1>
      <h2
        class="HvKpi-indicatorUnit e70l5p65 HvTypography-root HvTypography-title2 css-75jj3r-getStyledComponent-StyledIndicatorUnit e1tnpalo0"
      >
        Unit
      </h2>
    </div>
    <div
      class="HvKpi-comparisonComposition css-jr4b4k-StyledComparisonComposition e70l5p64"
    >
      <div>
        <div
          class="HvKpi-comparisonContainer css-1762388-StyledComparisonContainer e70l5p62"
        >
          <label
            class="HvKpi-comparisons HvKpi-spacingToTheRight e70l5p60 HvTypography-root HvTypography-label css-1fzhlyk-getStyledComponent-StyledInternalVisualComparisonTypography e1tnpalo0"
          >
            Visual comparison
          </label>
        </div>
        <div
          class="HvKpi-comparisonContainer css-1762388-StyledComparisonContainer e70l5p62"
        >
          <p
            class="HvKpi-comparisons HvTypography-root HvTypography-caption2 css-1s80glp-getStyledComponent e1tnpalo0"
          >
            Comparison indicator info
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
